public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts

```diff
@@ -232,6 +247,63 @@ class PanelCompatibilityWrapper {
     return [];
   }
 
+  public get targets() {
+    if (this._vizPanel.state.$data instanceof SceneQueryRunner) {
+      return this._vizPanel.state.$data.state.queries;
+    }
+
+    if (this._vizPanel.state.$data instanceof ShareQueryDataProvider) {
+      return [
+        {
+          datasource: {
+            uid: SHARED_DASHBOARD_QUERY,
+            type: 'datasource',
+          },
+          ...this._vizPanel.state.$data.state.query,
+        },
+      ];
+    }
+
+    if (this._vizPanel.state.$data instanceof SceneDataTransformer) {
+      if (this._vizPanel.state.$data.state.$data instanceof ShareQueryDataProvider) {
+        return [
+          {
+            datasource: {
+              uid: SHARED_DASHBOARD_QUERY,
+              type: 'datasource',
+            },
+            ...(this._vizPanel.state.$data.state.$data as ShareQueryDataProvider).state.query,
+          },
+        ];
+      }
+      if (this._vizPanel.state.$data.state.$data instanceof SceneQueryRunner) {
+        return (this._vizPanel.state.$data.state.$data as SceneQueryRunner).state.queries;
+      }
+    }
```

Comment:
maybe we should have optional getQueries() and getDataSource() on SceneDataProvider interface, that way SceneDataTransformer, and ShareQueryDataProvider can handle this this wrapping logic. 

this would then be 
```ts
 public get targets() {
   return this._vizPanel.state.$data.getQueries?.() ?? []; 
 } 
```

another option is to add an optional `getQueryRunner` to SceneDataProvider (SceneQueryRunner would just return this). 

But maybe a DashboardPanelDataProvider that handles this is the more simple solution